src/main/java/com/exasol/parquetio/data/GenericRow.java

```diff
@@ -30,11 +45,29 @@ public static GenericRow of(final Object... values) {
         return new GenericRow(Arrays.asList(values));
     }
 
+    /**
+     * A factory method to create a new instance with {@link MessageType} schema.
+     *
+     * @param values list of values
+     */
+    public static GenericRow of(final MessageType schema, final Object... values) {
+        return new GenericRow(schema, Arrays.asList(values));
+    }
+
     @Override
-    public Object getObjectAt(final int position) {
+    public Object getValue(final int position) {
         return values.get(position);
     }
 
+    @Override
+    public Object getValue(final String fieldName) {
```

Comment:
what happens when the field name is not defined?

Reply:
In case of an exception: Please also add a `bool has(String fieldName)` method since throwing and catching an exception is very slow in Java.

In  case of null: Not sure if it's a good idea. How can we distinguish between a null value and a nonexistent column?

Reply:
Would `isNull` method work for check if the the value is null? I think I have this in the original source.

Reply:
`isNullAt` is available going to add one for using with a column name.

Reply:
Added hasFieldName method.